### ISSUE TITLE
[member-ordering] arrow function doesn't recognize as function

### ISSUE THREAD

**Issue by MaksZhukov** [typescript-eslint/typescript-eslint#3271]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/member-ordering": [
        "error",
        { "interfaces": ["field", "method"] }
    ]
  },
    "parserOptions": {
    "babelOptions": {
      "configFile": "./config/babel.config.json"
    }
  },
}
```

```TS
interface Props {
  isCreatedByUser: boolean;
  onChange: (description: string) => void;
  value: string;
}
```

**Expected Result**

It has the wrong order and eslint throws an exception
```TS
onChange: (description: string) => void;
```

**Actual Result**

```TS
onChange: (description: string) => void;
```

It has the right order, but it's a function

**Additional Info**

Did eslint throw an exception?

No

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.29.0` |
| `@typescript-eslint/parser` | `2.0.0` |
| `TypeScript`                | `3.6.2` |
| `ESLint`                    | `5.16.0` |
| `node`                      | `14.14.0` |

**Comment by bradzacher** [comment]
> It has the wrong order and eslint throws an exception

Do you mean "reports an error"?

Could you please clarify what is happening and what you expect to happen? Your issue is very vague.

**Comment by MaksZhukov** [comment]
> > It has the wrong order and eslint throws an exception
> 
> You say "throws an exception" but further down you answered "no" to "did ESLint throw an exception?"
> Do you mean "reports an error"?
> 
> Could you please clarify what is happening and what you expect to happen? Your issue is very vague.
> 
> Please describe exactly what you expect to happen and exactly what is actually happening.

My point is that an arrow function is recognized as a field, not a method
So the order of the properties doesn't work correctly

**Comment by bradzacher** [comment]
> So the exception of the wrong order isn't thrown

Again you used the term "throw exception".
A thrown exception indicates a crash. A thrown exception is a big problem.

"Reporting an error" is *very* different.

Which is why I asked you to clarify what you mean exactly.

---

> My point is that an arrow function is recognized as a field, not a method

So you're saying that you expect that you expect that a function-typed property signature acts like a method?

A function-typed property signature is different to a method signature - they are treated differently by typescript (they have different variance). I don't think they should be seen as the same thing for the lint rule.

It's very different in comparison to a property with a value of an arrow-function vs a method.

**Comment by MaksZhukov** [comment]
> > So the exception of the wrong order isn't thrown
> 
> Again you used the term "throw exception".
> A thrown exception indicates a crash. A thrown exception is a big problem.
> 
> "Reporting an error" is _very_ different.
> 
> Which is why I asked you to clarify what you mean exactly.

Sorry, yes I meant "Reporting an error"

> > My point is that an arrow function is recognized as a field, not a method
> 
> So you're saying that you expect that you expect that a function-typed property signature acts like a method?
> 
> A function-typed property signature is different to a method signature - they are treated differently by typescript (they have different variance). I don't think they should be seen as the same thing for the lint rule.
> 
> It's very different in comparison to a property with a value of an arrow-function vs a method.

If they are different then is it possible to add a new member-type field like for arrow functions?

**Comment by bradzacher** [comment]
> If they are different then is it possible to add a new member-type field like for arrow functions?

No. Because you cannot syntactically detect every case, which creates an unpredictable lint rule.

Eg:
```ts
export type Func = () => void;

///

import type {Func} from './file';

class Foo {
  prop: Func; // cannot detect that this is a function type without type information
}
```
